src/YesSql.Core/Provider/IDbProviderOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using YesSql.Core.Services;
+
+namespace YesSql.Core.Provider
+{
+    public interface IDbProviderOptions
+    {
+        string ProviderName { get; set; }
```

Comment:
No, Orchard CMS works at a different level, once it knows what DbProvider and connection string to use, then it can call UseXXXServer(connectionString). It needs a list of available providers before this call. This is why Orchard "CMS" has its own providers extensibility.